src/methods/widen.js

```diff
@@ -0,0 +1,309 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Binding } from "../environment.js";
+import { FatalError } from "../errors.js";
+import type { Bindings, Effects, EvaluationResult, PropertyBindings, CreatedObjects, Realm } from "../realm.js";
+import type { Descriptor, PropertyBinding } from "../types.js";
+
+import { AbruptCompletion, PossiblyNormalCompletion } from "../completions.js";
+import { Reference } from "../environment.js";
+import { cloneDescriptor, equalDescriptors, IsDataDescriptor, StrictEqualityComparison } from "../methods/index.js";
+import { Generator } from "../utils/generator.js";
+import { AbstractValue, ObjectValue, Value } from "../values/index.js";
+
+import invariant from "../invariant.js";
+
+export class WidenImplementation {
+  _widenArrays(
+    realm: Realm,
+    v1: void | Array<Value> | Array<{ $Key: void | Value, $Value: void | Value }>,
```

Comment:
The type `Array<{ $Key: void | Value, $Value: void | Value }` appears all over the place. A descriptive alias would help readability

Reply:
I'm not sure that I can do better than "KeyValueList", which is about as readable as the actual type expression.